### PR TITLE
Add sequential values test

### PR DESCRIPTION
**Pull request by clrcrl**
This is a:
- [ ] bug fix PR with no breaking changes — please ensure the base branch is `master`
- [x] new functionality — please ensure the base branch is the latest `dev/` branch
- [x] (possibly) a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation
This macro builds on work done in #264, to add a general test that a column has sequential values (i.e. an arithmetic series)

It's a little different to the exact functionality of that test, but I think this one is more generally useful

🚨 Note: when getting the tests passing, I noticed that the `date_add` macro has some strange behavior on BigQuery — we use `datetime_add`, and cast to a datetime rather than a timestamp (thus losing any timezone info). I thought it would be preferential to use `timestamp_add` here instead, which meant I ended up touching some other files. @jtcohen6 — I'd really like your eyes on that part of the code

## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [ ] BigQuery
    - [ ] Postgres
    - [ ] Redshift
    - [ ] Snowflake
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [ ] I have added an entry to CHANGELOG.md

### CODE REVIEW

File: macros/schema_tests/sequential_values.sql [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{% macro test_sequential_values(model, column_name, interval=1, datepart=None) %}
```
Comment (jtcohen6) [inline macros/schema_tests/sequential_values.sql:1]: Given all the boiler-room work over in #312, could you add `dispatch` + `default__` here, too?

That takes nothing away from the fact that this is quite elegant SQL, and it runs beautifully on pg/rs/sf/bq :)

File: macros/cross_db_utils/dateadd.sql [hunk 2]
```diff
@@ -16,8 +16,8 @@
 
 {% macro bigquery__dateadd(datepart, interval, from_date_or_timestamp) %}
 
-        datetime_add(
-            cast( {{ from_date_or_timestamp }} as datetime),
+        timestamp_add(
+            cast( {{ from_date_or_timestamp }} as timestamp),
```
Comment (jtcohen6) [inline macros/cross_db_utils/dateadd.sql:20]: I'm not sure why we preferred `datetime` here. Looks like this change was made a long time ago (https://github.com/fishtown-analytics/dbt-utils/commit/85a615f2a6962febc8804b9aa7a9ce0d9945f38b).

Screen Shot 2021-01-06 at 12 51 49 PM

Agree that this is a _potentially_ breaking change—I think one worth making—and that we should make it in `datediff` too:
https://github.com/fishtown-analytics/dbt-utils/blob/ac78d49590ca04758d8ecdeb042eb8152bba6b0f/macros/cross_db_utils/datediff.sql#L17-L25

**Review by jtcohen6 — COMMENTED** [review]
I like it!

**Review by jtcohen6 — APPROVED** [review]
LGTM!

**Comment by clrcrl** [comment]
huh, this works locally, and worked before I cleaned up this branch 🤔

**Comment by clrcrl** [comment]
`:oof:`

```sql
Database Error in model test_dateadd (models/cross_db_utils/test_dateadd.sql)
  TIMESTAMP_ADD does not support the MONTH date part at [36:9]
```

OK leaving this for now, it's chewing up too much time

**Comment by clrcrl** [comment]
Closing in favor of #327

**Pull request closed without merging**